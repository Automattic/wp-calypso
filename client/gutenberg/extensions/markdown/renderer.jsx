/** @format */

/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { __ } from '@wordpress/i18n';
import { RawHTML } from '@wordpress/element';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm(
			__( 'Are you sure you wish to leave this page?', 'jetpack' )
		);

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
