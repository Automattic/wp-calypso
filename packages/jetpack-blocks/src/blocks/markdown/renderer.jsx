/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { RawHTML } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { __ } from '../../utils/i18n';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm( __( 'Are you sure you wish to leave this page?' ) );

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
