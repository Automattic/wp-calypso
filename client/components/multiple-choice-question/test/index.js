/** @format */
/**
 * External dependencies
 */
import { noop } from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import MultipleChoiceQuestion from '../';

describe( 'MultipleChoiceQuestion', () => {
	test( 'should render with the minimum required properties ( plus extra prop to guarantee order )', () => {
		const tree = renderer
			.create(
				<MultipleChoiceQuestion
					question={ 'Test Question One' }
					answers={ [
						{ id: 'test-answer-1', answerText: 'Test Answer One', doNotShuffle: true },
						{ id: 'test-answer-2', answerText: 'Test Answer Two', doNotShuffle: true },
						{ id: 'test-answer-3', answerText: 'Test Answer Three', doNotShuffle: true },
						{ id: 'test-answer-4', answerText: 'Test Answer Four', doNotShuffle: true },
					] }
					onAnswerChange={ noop }
				/>
			)
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
