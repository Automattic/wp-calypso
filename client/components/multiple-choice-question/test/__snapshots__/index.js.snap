// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleChoiceQuestion should render with the minimum required properties ( plus extra prop to guarantee order ) 1`] = `
<fieldset
  className="multiple-choice-question form-fieldset"
>
  <legend
    className="form-legend"
  >
    Test Question One
  </legend>
  <label
    className="form-label"
    locale="en"
  >
    <input
      checked={false}
      className="form-radio"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="test-answer-1"
    />
    <span>
      Test Answer One
    </span>
  </label>
  <label
    className="form-label"
    locale="en"
  >
    <input
      checked={false}
      className="form-radio"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="test-answer-2"
    />
    <span>
      Test Answer Two
    </span>
  </label>
  <label
    className="form-label"
    locale="en"
  >
    <input
      checked={false}
      className="form-radio"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="test-answer-3"
    />
    <span>
      Test Answer Three
    </span>
  </label>
  <label
    className="form-label"
    locale="en"
  >
    <input
      checked={false}
      className="form-radio"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="test-answer-4"
    />
    <span>
      Test Answer Four
    </span>
  </label>
</fieldset>
`;
